var/spack/repos/builtin/packages/root/package.py

```diff
@@ -153,6 +153,8 @@ class Root(CMakePackage):
             sha256="e68be5fe7b1ec873da134bd39c5c72730c4ca06d51b52eb436ae44fe81cd472d",
             when="@:6.30.04 +x",
         )
+        # Fix rpath for loading cppyy
+        patch("loader.patch", when="@6.28: +python")
```

Comment:
```suggestion
        patch(
            "https://github.com/root-project/root/pull/15925.diff?full_index=1",
            sha256="1937290a4d54cd2e3e8a8d23d93b8dedaca9ed8dcfdcfa2f0d16629ff53fb3b7",
            when="@6.28: +python"
        )
```
Should this apply starting with 6.26? Last change to the patch context is at https://github.com/root-project/root/commit/64e8de57cea981f856971dd9d934f4881deee771.

Reply:
will have to x-check when I find a moment but I believe root 6.26 is kind of too old to be compilable on my Mac, so I might not be able to test that actually...

Reply:
I see that the `root` package has:
```
    # See https://github.com/root-project/root/issues/11714
    if sys.platform == "darwin" and macos_version() >= Version("13"):
        conflicts("@:6.26.09", msg="macOS 13 support was added in root 6.26.10")
```
I don't know when macOS 12 reaches end of life, but it seems only 6.26.10 and 6.26.14 would be under consideration here.

If we we start the patch at 6.28, maybe we can alternatively just add a conflict with older version for any macos version.

Reply:
Added the last comment here as conflict. Now, on macOS 12, the patch is either applied (`@6.28: +python`) or a conflict is thrown (`@:6.27 +python`).